backend/de.metas.postfinance/src/main/java/de/metas/postfinance/docoutboundlog/PostFinanceLogRepository.java

```diff
@@ -70,11 +71,12 @@ private PostFinanceLog toPostFinanceLog(@NonNull final I_C_Doc_Outbound_Log_Post
 				.build();
 	}
 
-	public Optional<PostFinanceLog> retrieveLatestLogWithTransactionId(@NonNull final String transactionId)
+	public Optional<PostFinanceLog> retrieveLatestLogWithTransactionIdAndOrgId(@NonNull final String transactionId, @NonNull final OrgId orgId)
 	{
 		return queryBL.createQueryBuilder(I_C_Doc_Outbound_Log_PostFinance_Log.class)
 				.addOnlyActiveRecordsFilter()
-				.addEqualsFilter(I_C_Doc_Outbound_Log_PostFinance_Log.COLUMN_PostFinance_Transaction_Id, transactionId)
+				.addStringLikeFilter(I_C_Doc_Outbound_Log_PostFinance_Log.COLUMNNAME_PostFinance_Transaction_Id, transactionId, false)
+				.addEqualsFilter(I_C_Doc_Outbound_Log_PostFinance_Log.COLUMNNAME_AD_Org_ID, orgId)
```

Comment:
I think we don't need the OrgId here, because the transactionId contains <InvoiceId/DunningId><DocBaseType><DocSubType><DocumentNo> and because of this should be unique

Reply:
Because this is the earliest steps where I can exclude entries that are not for the parameter org.

Reply:
You already filtered by org, when you request processResults with orgs BillerId. And if you filter the results after download, if 2 Orgs use the same BillerId ou aren't able to download them again without changing isArchive

Reply:
right. Removed the org parameter.